internal/scheduled/runner.go

```diff
@@ -73,45 +73,49 @@ func NewRunner(logger Logger, store ScheduledStore, donorStore DonorStore, notif
 
 // Run the Runner, it is expected to be called in a Go routine.
 func (r *Runner) Run(ctx context.Context) error {
-	ticker := time.Tick(r.period)
-
-	for {
-		innerCtx, cancel := context.WithTimeout(ctx, r.period)
-		defer cancel()
-
-		r.logger.InfoContext(ctx, "runner step started")
-		if err := r.step(innerCtx); err != nil {
-			r.logger.ErrorContext(ctx, "runner step error", slog.Any("err", err))
-		}
-		r.logger.InfoContext(ctx, "runner step finished")
-
-		select {
-		case <-ctx.Done():
-			return nil
-		case <-ticker:
-			continue
-		}
+	//ticker := time.Tick(r.period)
+	//
+	//for {
+	//	innerCtx, cancel := context.WithTimeout(ctx, r.period)
+	//	defer cancel()
+
+	r.logger.InfoContext(ctx, "runner step started")
+	if err := r.step(ctx); err != nil {
+		r.logger.ErrorContext(ctx, "runner step error", slog.Any("err", err))
+		return err
 	}
+	r.logger.InfoContext(ctx, "runner step finished")
+
+	//	select {
+	//	case <-ctx.Done():
+	//		return nil
+	//	case <-ticker:
+	//		continue
+	//	}
+	//}
+	return nil
 }
 
 func (r *Runner) step(ctx context.Context) error {
-	r.waiter.Reset()
+	//r.waiter.Reset()
+	//r.logger.InfoContext(ctx, "waiter reset")
 
 	for {
 		row, err := r.store.Pop(ctx, r.now())
 		if errors.Is(err, dynamo.NotFoundError{}) {
+			r.logger.InfoContext(ctx, "not found")
+
 			return nil
 		} else if errors.Is(err, dynamo.ConditionalCheckFailedError{}) {
 			r.logger.InfoContext(ctx, "runner conditional check failed")
-			if err := r.waiter.Wait(); err != nil {
```

Comment:
you'll want some kind of waiting, otherwise it may just keep hammering dynamo when there is an underlying problem